src/orange/OrangeTypes.hh

```diff
@@ -33,6 +33,9 @@ using logic_int = unsigned short int;
 //! Identifier for a face local to a particular volume (internal use only)
 using FaceId = OpaqueId<struct Face>;
 
+//! Identifier for the current "level", i.e. depth of embedded universe
+using LevelId = OpaqueId<unsigned short int>;
```

Comment:
```suggestion
using LevelId = OpaqueId<struct Level, unsigned short int>;
```

Reply:
What is the reasoning for this?

Reply:
The first template argument is the "instantiator", which for most contexts is the item type being accessed. For abstractions though we use "struct X" to declare a nonexistent but meaningful standin. I assume you meant unsigned short to be the integer capacity?

Reply:
Looks like this should actually be just

```
using LevelId = OpaqueId<struct Level>;
```
?

Reply:
Probably. I thought that you were deliberately trying to use a smaller integer size to reduce storage/bandwidth, because we know the level will always be super duper small.